### PR TITLE
ECS verbs and update context menu

### PR DESCRIPTION
**Pull request by ElectroJr**
# About this PR

The main point of this PR is to move verbs away from being classes defined in components, and instead make them defined in systems via event subscriptions. For a concise example of how verbs are now defined, see the toggle light verb in `HandHeldLightSystem.cs`.

These changes required making changes to the mess that is `ContextMenuPresenter` and the verb menu. I've tried to make them a bit more sensible and update the UI a bit, but they are still a mess. Some more verb-display options were also added (e.g., color text, tooltips for disabled verbs, an icon-only option).

I've also added a couple of new verbs
- Eject verbs now have an Insert counterpart
- You can now buckle pulled/held entities to a chair/bed (previously only unbuckle)
- Add Gas & Set Temperature verbs, which just open the old UI but with coordinates set to the  the clicked entity's coordinates.

Previously, entities would disappear from the right click context menu if they moved more than 1 tile from the clicked location. This made it impossible to interact with yourself or other entities while they were moving. This has been changed, so that entities are only removed from the menu when you loose sight of them.

## ECS Verbs Overview

For an overview of how:
- Verbs are defined, see the verb in `HandHeldLightSystem`
- Verbs can be "collected" with an event and executed, see `InteractionSystem.AltInteract()`

Previously verbs were classes defined in components, that inherited from the verb class. The class would have a function to get verb data (text, icon, visibility, disabled), and a separate function to run the verb.  The verb system would iterate over all components on a target entity, look for any verbs, and add them to a list.

Now verbs are just instances of the verb class. Instances have information about the verb (priority, text, icon, disabled) and execution instructions (an action delegate, or a local event to raise on execution).  The verb system raises a GetVerb event directed at an entity. Systems can subscribe to that event, and add their verbs to a sorted list.

### Verb/interaction types
There are several different GetVerbEvents, used to assemble lists of different types of verbs:
- Interaction, correspond to verbs whose effects can also be triggered by left-clicking or using 'Z'
- Activation, correspond to verbs whose effects can also be triggered by using 'E' (or left-click/Z if no other interaction takes place)
- Alternative, correspond to verbs whose effects can be triggered by using alt + left-click/Z/E
- Other, for miscellaneous verbs like examine, debug, deconstruct.

Except for the alternative verbs, which are how alt-clicking interactions are actually defined, there is currently no functional difference between these verb types other than the order in which they are displayed. __**Currently**__, the "other", "activation" and "interaction" verb assembly events could just be merged into one event.

The reason for distinguishing between these verb types is so that the verbs can be visually differentiated in the context menu. This would allow users to discover what interactions are possible on an entity, and what keybindings should be used to trigger them.  Currently, the context menu only uses the verb types to determine ordering. The displayed verbs are sorted first by verb-type, and then by verb priority. However, it is trivial to change the color of the verb text or button based on the verb type. I haven't done so, because I have no idea what colors to choose, input is appreciated. Personally I would just prefer bold/italic/underlined text, but the RichTextLabel does not support that.

## Changed Files

The main verb related files that have been changed include:
- Client, server, and shared `VerbSystem`
- `Verb.cs`
- `VerbCategory.cs`
- `VerbEvents.cs`
- `VerbMenuElement.cs`
- `ContextMenuElement.cs`
- `ContextMenuPresenter`

Several un-ECS-ed components had corresponding system files created to handle new verbs. 
Several component functions were changed from `private` -> `public`. In some instances `TryDoX()` functions were split into `CanDoX()`, `DoX()`, and `TryDoX()`, to facilitate adding verbs only if they are actually executable.

These changes resulted in empty server and client PullingComponents, so these were removed and `SharedPullingComponent` was renamed to just `PullingComponent`.

## UI Changes

- The verb menu and the context menu now use the same colors, The colors have been slightly changed and larger margins were added. The verb ordering has also changed.
- The old context menu had a scroll-bar that appeared when more than 10 items were listed, but this cropped the entity text and was hard to see. This has been fixed, and scrollbar support was added to the verb menu (which previously did not have it).
- Verbs can now be shown as icons only, left-to-right. This was inspired by PJB's MS paint concept art masterpiece ()

### Images

Changes |Old| New
-----|----| -----
General Look |  |   
Scrollbar |  |   
Inline rotate |   | 

### Duplicate code

Some systems/components only provide a single monolithic `TryDo` function that I haven't split into `CanDo/Do/TryDo`. As a result the "should we show this verb" checks involve duplicate code, and both bits of code have to be kept consistent. This issue isn't new, it already exists for the current verbs. In future this should be changed. In the worst cases, the verbs do those checks, and then run them again when executing using the provided `TryDo()` function.

Additionally, there are many eject/insert verbs. There should be some sort of system that manages general  insert/eject item slots, so that these verbs only have to be defined once. To a lesser extent, this is also a problem for "open UI" verbs.

### Context & Verb UI

The verb & entity context menu UIs are still a mess. In a seperate branch I've moved it to use XAML & stylesheets, as well as make the context & verb menu nicer to read & use. I can merge that into this branch or leave it for later.

### IEntity vs EntityUID

The verb events & system currently still use IEntity instead of uids. This is mainly because many of the existing interaction functions expect IEntity as arguments, given that is what the various interaction events use.

Modifying all those existing functions, or adding many uid->entity resolution calls so that at-least the verb system/events use uids can be done later.

# Changelog

:cl:
- add: Added several new verbs for inserting objects and (un)buckling entities.
- tweak: Some verbs can now be triggerd by alt clicking (e.g., spin revolver). Others have had alt-click removed (e.g., open storage UI), because that functionality is already provided by the 'E' keybind.
- fix: Moving entities remain in the context menu until out of view (you can now interact with yourself while moving).

### CODE REVIEW

File: Content.Client/ContextMenu/UI/ContextMenuElement.cs [hunk 1]
```diff
@@ -229,15 +231,18 @@ protected override Vector2 MeasureOverride(Vector2 availableSize)
                 return Vector2.Zero;
             }
 
-            List.Measure(availableSize);
-            var listSize = List.DesiredSize;
+            Scroll.Measure(availableSize);
+            var size = List.DesiredSize;
```
Comment (ElectroJr) [inline Content.Client/ContextMenu/UI/ContextMenuElement.cs:235]: I'll probabyl be undoing the GUI scrollbar changes for now, but still want to know: why do ScrollContainers not return desired-size information? Is there any way to easily know how much vertical  space will be taken up by the scrollbar?

File: Content.Server/Administration/AdminVerbSystem.cs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+
+namespace Content.Server.Administration
+{
+    /// <summary>
+    ///     System to provide various global admin/debug verbs
+    /// </summary>
+    public class AdminVerbSystem : EntitySystem
```
Comment (ElectroJr) [inline Content.Server/Administration/AdminVerbSystem.cs:29]: For now I've just placed most of the various admin debug-verbs into this system, despite the fact that some of them could also just go into their own systems (e.g., control mob and make sentient could go into the mind system).

I'm not sure which is preferred. Either way, this system would still have to exist for some verbs that have no other clearly associated system (e.g., "delete" & "Rejuvenate").

File: Content.Server/Administration/AdminVerbSystem.cs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+        private void AddDebugVerbs(GetOtherVerbsEvent args)
+        {
+            if (!args.User.TryGetComponent<ActorComponent>(out var actor))
+                return;
+
+            var player = actor.PlayerSession;
+
+            // Delete verb
+            if (_groupController.CanCommand(player, "deleteentity"))
```
Comment (ElectroJr) [inline Content.Server/Administration/AdminVerbSystem.cs:50]: Currently the way that admin command checks were done was very inconsistent. They usually used either `CanCommand`, `CanAdminMenu`, or `HasAdminFlag()`. Which one(s) should be used?

File: Content.Server/Verbs/VerbSystem.cs [hunk 4]
```diff
@@ -24,157 +26,110 @@ public override void Initialize()
...
-                }
-
-                verb.Activate(userEntity, component);
-                break;
+                return;
             }
 
-            foreach (var globalVerb in VerbUtility.GetGlobalVerbs(Assembly.GetExecutingAssembly()))
+            var verbs = GetVerbs(targetEntity, userEntity, args.Type)[args.Type];
+            var verb = verbs.Where((v) => v.Key == args.VerbKey).FirstOrDefault();
```
Comment (mirrorcult) [inline Content.Server/Verbs/VerbSystem.cs:85]: iirc this kind of LINQ is really slow

File: Content.Shared/Verbs/HideContextMenuComponent.cs [hunk 5]
```diff
@@ -1,15 +0,0 @@
-﻿using Robust.Shared.GameObjects;
-
-namespace Content.Shared.Verbs
-{
-    [RegisterComponent]
-    public class HideContextMenuComponent : Component, IShowContextMenu
```
Comment (mirrorcult) [inline Content.Shared/Verbs/HideContextMenuComponent.cs:6]: why was this removed? its used on some stuff, like low wall entities

File: Content.Shared/Verbs/SharedVerbSystem.cs [hunk 6]
```diff
@@ -1,62 +1,139 @@
...
+        }
+
+        /// <summary>
+        ///     Raises a number of events in order to get all verbs of the given type(s)
+        /// </summary>
+        public Dictionary<VerbType, List<Verb>> GetVerbs(IEntity target, IEntity user, VerbType verbTypes)
+        {
+            Dictionary<VerbType, List<Verb>> verbs = new();
+
+            if (verbTypes.HasFlag(VerbType.Activation))
```
Comment (mirrorcult) [inline Content.Shared/Verbs/SharedVerbSystem.cs:70]: don't use HasFlag, just do `verbTypes & VerbType.Activation` or whatever bitwise operation you need

https://stackoverflow.com/questions/7368652/what-is-it-that-makes-enum-hasflag-so-slow

File: Content.Shared/Verbs/Verb.cs [hunk 7]
```diff
@@ -1,90 +1,143 @@
-﻿using System;
-using JetBrains.Annotations;
 using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+using System;
 
 namespace Content.Shared.Verbs
 {
+    [Flags] public enum VerbType
```
Comment (mirrorcult) [inline Content.Shared/Verbs/Verb.cs:8]: separate lines for the attribute declaration

File: Content.Shared/Verbs/Verb.cs [hunk 8]
```diff
@@ -1,90 +1,143 @@
...
+    ///     events, or networked events. Verbs also provide text, icons, and categories for displaying in the
+    ///     context-menu.
     /// </summary>
-    /// <remarks>
-    ///     To add a verb to an entity, define it as a nested class inside the owning component,
-    ///     and mark it with <see cref="VerbAttribute"/>
-    /// </remarks>
-    [UsedImplicitly]
-    public abstract class Verb : VerbBase
+    [Serializable, NetSerializable]
```
Comment (mirrorcult) [inline Content.Shared/Verbs/Verb.cs:22]: why is this serializable/netserializable? i dont think this can be sent over the wire (esp. since delegates and events cant be serialized?)

File: Content.Shared/Verbs/VerbCategory.cs [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+            new("verb-categories-open", "/Textures/Interface/VerbIcons/open.svg.192dpi.png", true);
+
+        public static readonly VerbCategory Close =
+            new("verb-categories-close", "/Textures/Interface/VerbIcons/close.svg.192dpi.png", true);
+
+        public static readonly VerbCategory Rotate =
+            new("verb-categories-rotate", "/Textures/Interface/VerbIcons/refresh.svg.192dpi.png", iconsOnly: true);
+
+        public static readonly VerbCategory SetTransferAmount =
+            new("verb-categories-transfer", "/Textures/Interface/VerbIcons/spill.svg.192dpi.png");
```
Comment (mirrorcult) [inline Content.Shared/Verbs/VerbCategory.cs:84]: this is pretty ugly.. honestly might be nice if verb categories were prototypes? i'm not sure

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Administration/AdminVerbSystem.cs (hunk 3 above)
Comment (mirrorcult) [reply Content.Server/Administration/AdminVerbSystem.cs:50]: CanCommand makes sense to me here

File: Content.Client/ContextMenu/UI/ContextMenuElement.cs (hunk 1 above)
Comment (mirrorcult) [reply Content.Client/ContextMenu/UI/ContextMenuElement.cs:235]: @PJB3005

File: Content.Server/Access/IdCardConsoleSystem.cs [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+                !args.CanAccess ||
+                !args.CanInteract ||
+                !args.Using.HasComponent<IdCardComponent>() ||
+                !_actionBlockerSystem.CanDrop(args.User))
+                return;
+
+            // Can we insert a privileged ID? 
+            if (component.PrivilegedIDEmpty)
+            {
+                Verb verb = new("IDConsole:insertPrivileged");
```
[the remark this answers is not in the document]
Comment (mirrorcult) [reply Content.Server/Access/IdCardConsoleSystem.cs:33]: is it even strictly necessary?

**Comment by mirrorcult** [comment]
one issue im noticing ingame is that verb text isnt centered properly so the icon and text look weird next to eachother 
image

File: Content.Server/Access/IdCardConsoleSystem.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Access/IdCardConsoleSystem.cs:33]: There is no actual requirement other than they should be unique. And yeah I was probably quite inconsistent with how I defined them while working on the PR. 

The keys are used to identify verbs when the client asks the sever to run a verb from the context menu that was defined server-side. So currently they are required, but maybe there is some better way of informing the server of what verb the client wants to run.

Previously when verbs were just classes these keys were generated automatically when the verbs were discovered while iterating over components using:
`$"{component.GetType()}:{verb.GetType()}"`
This lead to long (but at least guranteed to be unique) keys:
"Content.Server.Inventory.Components.InventoryComponent:Content.Server.Inventory.Components.InventoryComponent+SetOutfitVerb"

Now that verbs are not individual classes with unique names I don't know if there is a nice way of automatically generating unique keys. But at the very least what I've should be changed to be consistent.

File: Content.Shared/Verbs/HideContextMenuComponent.cs (hunk 5 above)
Comment (ElectroJr) [reply Content.Shared/Verbs/HideContextMenuComponent.cs:6]: I just changed this to be a tag instead. I can undo that and keep it as a component?

File: Content.Shared/Verbs/Verb.cs (hunk 8 above)
Comment (ElectroJr) [reply Content.Shared/Verbs/Verb.cs:22]: Previously there was a dedicated serializable `NetVerbData` struct that had fields for the verb text, key, category, icon, and visibility. This was used to add server-side verbs to the the client's context menu.

At some point I realised that Verb and NetverbData were basically identical except for the delegate & events. So I just made Verb serializable and marked the delegate & events with the [NonSerialized] attribute.

As far as I'm concerned, the netcode may as well be black magic. So maybe I wasn't meant to do that but it seems to work and just doesn't send those nullable fields to the client.

File: Content.Shared/Verbs/VerbCategory.cs (hunk 9 above)
Comment (ElectroJr) [reply Content.Shared/Verbs/VerbCategory.cs:84]: I agree that  these should maybe just be prototypes. It should also reduce network bandwidth when sending verbs over the network (just the prototype id, instead of full object with text & SpriteSpecifier).

On the other hand, you would then be hardcoding a bunch of yaml prototype IDs into a lot of systems and that feels icky.

File: Content.Shared/Verbs/SharedVerbSystem.cs (hunk 6 above)
Comment (ElectroJr) [reply Content.Shared/Verbs/SharedVerbSystem.cs:70]: Changed. Also fixed a Activation vs Interaction typo

File: Content.Server/Access/IdCardConsoleSystem.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Access/IdCardConsoleSystem.cs:33]: For now I'll just make it consistently use PascalCase with "[component/system]:[verbname/description]"

File: Content.Server/Chemistry/EntitySystems/SolutionTransferSystem.cs [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+            if (!args.CanAccess || !args.CanInteract || !component.CanChangeTransferAmount)
+                return;
+
+            if (!args.User.TryGetComponent<ActorComponent>(out var actor))
+                return;
+            
+            // Custom transfer verb
+            Verb custom = new("transfer:custom");
+            custom.Text = Loc.GetString("comp-solution-transfer-verb-custom-amount");
+            custom.Category = VerbCategory.SetTransferAmount;
```
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Chemistry/EntitySystems/SolutionTransferSystem.cs:45]: Currently categories are just instances of the VerbCategory class. There are just several static instances defined in the VerbCategory class itself. You can always create a new verb category instance when defining your verb, so they are already dynamic. But yeah, I don't know if there's a need for that.

The way they are done at the moment also means that instead of just sending a verb category key, each verb in the list has it's category instance serialized and sent over the network, which is wasteful if dynamic categories are never used.

File: Content.Shared/Verbs/HideContextMenuComponent.cs (hunk 5 above)
Comment (mirrorcult) [reply Content.Shared/Verbs/HideContextMenuComponent.cs:6]: tags fine i suppose

File: Content.Server/Items/ItemSystem.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+                !args.CanInteract ||
+                !component.CanPickup(args.User, popup: false))
+                return;
+
+            Verb verb = new("pickup");
+            verb.Act = () => args.Hands.TryPutInActiveHandOrAny(args.Target);
+            verb.IconTexture = "/Textures/Interface/VerbIcons/pickup.svg.192dpi.png";
+
+            // if the item already in the user's inventory (or some other storage), change the text
+            if (args.Target.IsInContainer())
```
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Items/ItemSystem.cs:31]: Yes, entities in containers are not shown on the context menu unless the user is also in the same container. However, currently you can still right-click on your hands or inventory slots directly. This bypasses the context menu and directly opens up the verb menu for the clicked entity. 

The text change was made so that you don't "pick up" and item in your pocket, or backpack, you just place it into your hands. But now I realise it also changes the text in other situations (e.g., the pickup text is changed for picking up items in a locker, while you are also in that locker, see picture). I guess I'll just change it from checking if it is in **any** container to checking if the user **IS** the container. Or just get rid of the text change, its completely unnecessary.

Screenshot_13

File: Content.Server/Verbs/VerbSystem.cs (hunk 4 above)
Comment (ElectroJr) [reply Content.Server/Verbs/VerbSystem.cs:85]: Yeah this really should have just been a for loop.

File: Content.Server/Items/ItemSystem.cs (hunk 12 above)
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Items/ItemSystem.cs:31]: Changed to
```cs
if (args.Target.TryGetContainer(out var container) && container.Owner == args.User)
```

File: Content.Client/Verbs/VerbMenuElement.cs [hunk 13]
```diff
@@ -0,0 +1,232 @@
...
+
+            AddChild(buttonContents);
+
+            if (Disabled)
+                return;
+
+            // give the button functionality!
+            OnPressed += _ =>
+            {
+                system.ContextMenuPresenter.CloseAllMenus();
```
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Client/Verbs/VerbMenuElement.cs:103]: I've added that option. Currently only used for rotate, but I guess it could eventually also be useful for some other actions like knocking on windows or maybe attacking, generally whenever you want to repeatedly do the same action.

File: Content.Server/Cabinet/ItemCabinetSystem.cs [hunk 14]
```diff
@@ -1,33 +1,88 @@
...
+
             SubscribeLocalEvent<ItemCabinetComponent, InteractUsingEvent>(OnInteractUsing);
             SubscribeLocalEvent<ItemCabinetComponent, InteractHandEvent>(OnInteractHand);
             SubscribeLocalEvent<ItemCabinetComponent, ActivateInWorldEvent>(OnActivateInWorld);
 
             SubscribeLocalEvent<ItemCabinetComponent, TryEjectItemCabinetEvent>(OnTryEjectItemCabinet);
             SubscribeLocalEvent<ItemCabinetComponent, TryInsertItemCabinetEvent>(OnTryInsertItemCabinet);
             SubscribeLocalEvent<ItemCabinetComponent, ToggleItemCabinetEvent>(OnToggleItemCabinet);
+
+            SubscribeLocalEvent<ItemCabinetComponent, GetInteractionVerbsEvent>(AddCabinetVerbs);
```
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Cabinet/ItemCabinetSystem.cs:38]: Yes. there was also one for the backpack / item storage component to open the UI. I forgot to mention that, and will add that to the changelog.

As to why: I realised that both of these actions can always executed via the 'E' key, so having them be alt-interactions was redundant.  Also for the purpose of distinguishing interaction types in the verb menu according to what key-binds can trigger the actions, these verbs would have needed to belong to two different verb categories simultaneously.

However I did make a mistake here, the open/close should be 'Activation' not 'Interaction' verbs, so I'll change that.

File: Content.Server/Access/IdCardConsoleSystem.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (mirrorcult) [reply Content.Server/Access/IdCardConsoleSystem.cs]: is there any runtime verification/a test that they're unique?

[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Server/Access/IdCardConsoleSystem.cs]: No there wasn't.  I've now made changes to remove verb keys altogether. Instead of using a `List<Verb>` & the `List.Sort()` function, it is now just using a `SortedSet<Verb>`. The "verb keys" are effectively implicitly defined by the verb `IComparable` function. 

If a new verb is equivalent according to the comparator, it will not be added to the set. Given that the comparator uses the verb text, icon, category and priority to sort the set of verbs, in most instances this should also guarantee that you never get entries in the context menu that appear identical.

This basically just shifts the "you should use unique keys" requirement to "you should use a unique verb text & category combination". Gven that requiring descriptive context menu entries is a sensible requirement that should be imposed anyways, I think that is fine.

[the remark this answers is not in the document]
Comment (mirrorcult) [reply Content.Server/Access/IdCardConsoleSystem.cs]: epic

File: Content.Client/Verbs/AdminVerbSystem.cs [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+            // Verb to open Set temperature admin window, with pre-filled coordinates
+            if (_clientConGroupController.CanCommand("settemp"))
+            {
+                Verb verb = new();
+                verb.Category = VerbCategory.Debug;
+                verb.Text = "Set Temperature";
+                // TODO VERB ICON add thermometer?
+                verb.Act = () =>
+                {
+                    var window = _dynamicTypeFactory.CreateInstance<SetTemperatureWindow>();
```
Comment (mirrorcult) [inline Content.Client/Verbs/AdminVerbSystem.cs:72]: uhh... did it use dynamictypefactory before? this should be using eUI imo

Comment (ElectroJr) [reply Content.Client/Verbs/AdminVerbSystem.cs:72]: It was doing this before, I just copied it across.  I don't even know why this used dynamic type factory anyways. I don't really know when the type factory would be needed but this function apparently just:
> Constructs a new instance of the given type with Dependencies resolved.

Given that this window has no dependencies, AFAIK this could just be a normal constructor, and it still seems to work fine without the dynamic type factory

Comment (ElectroJr) [reply Content.Client/Verbs/AdminVerbSystem.cs:72]: More details: I got this code from `Content.Client.Administration.UI.CustomControls.UICommandButton`. Previously this window was only opened by pressing this ui button in the admin menu, and I just copied over the same window opening code.

So I guess the reason it was using dynamic type factory is because at least one of those admin menus had dependencies. Now I'm not sure whether to change that code so it no longer uses it or keep it as is.

Comment (ElectroJr) [reply Content.Client/Verbs/AdminVerbSystem.cs]: These, and the other new verb can be added sometime later in a seperate PR, removed for now.

**Comment by ElectroJr** [comment]
Is good to review. 

On the UI/XAML TODO: I've spent some time making the entity and verb context menus use XAML & style-sheets, as well just re-writing them in general so that the UI logic is separated from the verb system.  However that makes this pr even bigger (from ~ +3300/-4300 line changes to ~ +4000/-5000). So unless people would rather just rip off the verb UI bandaid in a single PR, that can wait for a separate PR.

File: Content.Client/Verbs/VerbMenuElement.cs [hunk 16]
```diff
@@ -0,0 +1,232 @@
...
+
+    public sealed class VerbButton : BaseButton
+    {
+        public VerbButton(VerbSystem system, Verb verb, VerbType type, EntityUid target, bool drawIcons = true) : base()
+        {
+            Disabled = verb.Disabled;
+            ToolTip = verb.Tooltip;
+            TooltipDelay = 0.5f;
+
+            var buttonContents = new BoxContainer { Orientation = LayoutOrientation.Horizontal };
```
[the remark this answers is not in the document]
Comment (ElectroJr) [reply Content.Client/Verbs/VerbMenuElement.cs:63]: I can merge XAMl & other UI changes into this pr, but its probably better to leave as is for now?

[the remark this answers is not in the document]
Comment (mirrorcult) [reply Content.Client/Verbs/VerbMenuElement.cs:63]: if you have them already done and just stashed then we can leave them for another pr tbh

**Review by mirrorcult — APPROVED** [review]
